plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -658,10 +659,9 @@ private void addSplitPluginDependencies(String thisPlugin, MavenRunner.Config mc
                     System.out.println("Skipping implicit dep " + thisPlugin + " → " + plugin);
                     continue;
                 }
-                VersionNumber splitPoint = new VersionNumber(pieces[1]);
+                String splitPoint = pieces[1];
                 VersionNumber declaredMinimum = new VersionNumber(pieces[2]);
-                // TODO this should only happen if the tested core version is ≥ splitPoint
-                if (coreDep.compareTo(splitPoint) <= 0 && !pluginDeps.containsKey(plugin)) {
+                if (coreCoordinates.compareVersionTo(splitPoint) >= 0 && !pluginDeps.containsKey(plugin)) {
```

Comment:
Are you saying that is should look like:
`if (coreCoordinates.compareVersionTo(splitPoint) >= 0 && coreDep.compareTo(splitPoint) >= 0 && !pluginDeps.containsKey(plugin))`

The problems I've seen that are related to/fixed by this are around a plugin not loading due to an outdated dependency of a plugin that was bundled at one point.  The ones that are no longer bundled should be upgraded like the other dependencies.